pkg/protocol/commitment_verifier.go

```diff
@@ -58,27 +75,49 @@ func (c *CommitmentVerifier) verifyCommitment(commitment *model.Commitment, atte
 		return nil, 0, ierrors.Errorf("invalid merkle proof for attestations for commitment %s", commitment.ID())
 	}
 
-	// 2. Verify attestations.
+	// 2. Update validatorAccountsData if fork happened across epoch boundaries.
+	//    We try to use the latest accounts data (at lastCommonSlotBeforeFork) for the current epoch.
+	//    This is necessary because the committee might have rotated at the epoch boundary and different validators might be part of it.
+	//    In case anything goes wrong we keep using previously known accounts data (initially set to the accounts data
+	//    of the validators for the epoch of the last common commitment before the fork).
+	apiForSlot := c.engine.APIForSlot(commitment.Slot())
+	commitmentEpoch := apiForSlot.TimeProvider().EpochFromSlot(commitment.Slot())
+	if commitmentEpoch > c.epoch {
+		c.epoch = commitmentEpoch
+
+		committee, exists := c.engine.SybilProtection.SeatManager().CommitteeInEpoch(commitmentEpoch)
+		if exists {
+			validatorAccounts, err := committee.Accounts()
+			if err == nil {
+				validatorAccountsData, err := c.engine.Ledger.PastAccounts(validatorAccounts.IDs(), c.lastCommonSlotBeforeFork)
```

Comment:
So if you have a committee for the new epoch you take those accounts with the data at the last common slot before the fork. If instead you don't have a committee for the new epoch you are going to keep using the previous (or initial) `validatorAccountsData`; you are therefore assuming that the committee should be re-used and it was not rotated.
If I reed the following code correctly, in either case you would need accurate account data at `lastCommonSlotBeforeFork` _**ONLY** in terms of signing keys_: to verify attestations' signatures.

Reply:
Yes exactly